spec/controllers/api/agencies_controller_spec.rb

```diff
@@ -89,4 +96,20 @@ def expected_response
       ]
     }
   end
+
+  def get_coordinates(zip_query, latitude, longitude, request_params)
+    return '' if request_params.to_s.include?(':zip_code') == false
+
+    zip_lat = ::ZipCode.select(:latitude)
+                       .where(zip_code: zip_query).to_a.first.latitude
+    zip_long = ::ZipCode.select(:longitude)
+                        .where(zip_code: zip_query).to_a.first.longitude
+    calc_distance(zip_lat, zip_long, latitude, longitude)
+  end
+
+  def calc_distance(zip_lat, zip_long, latitude, longitude)
+    Geocoder::Calculations.distance_between([zip_lat, zip_long],
+                                            [latitude,
+                                             longitude]).round(2)
+  end
```

Comment:
This test looks like it's just duplicating the source code. If there's a bug in the code, these tests will just duplicate the error. Rather than do it this way, it would be better to move the distance calculation into a model and directly test the distance from some known values.

Reply:
I'd like to learn more.  Let's discuss in slack.

Reply:
The distance in the expected result is now derived by calling the event model method with the coordinates of the zip_code parameter.